paddle/platform/place.h

```diff
@@ -32,17 +32,23 @@ struct CPUPlace {
 };
 
 struct CUDAPlace {
+  enum StreamType { kCOMPUTATION, kCOMMUNICATION, kLast };
```

Comment:
Our original idea of `Place` is to represent the memory location of a tensor. I am not sure it's a good idea to add cuda stream into `Place`.

Reply:
We need one to one mapping between CUDAPlace and CUDAContext. If we have CUDAPlace mapped to multiple CUDACtx, how should the API of `DeviceContextPool::Get(const platform::Place& place)` look like?